### ISSUE TITLE
Explanations using `add_internal` and `union_trusted`

### ISSUE THREAD

**Issue by dewert99** [egraphs-good/egg#282]
The `EGraph::union_trusted` method currently has the documentation:

> Unions two e-classes, using a given reason to justify it.
>
> Unlike union_instantiations, this function picks arbitrary representatives from either e-class. When possible, use [union_instantiations](https://docs.rs/egg/latest/egg/struct.EGraph.html#method.union_instantiations), since that ensures that the proof rewrites between the terms you are actually proving equivalent.

As far as I can tell this "arbitrariness" comes from the fact that publicly exposed methods for getting `Id`s from the `EGraph` (`add`, `add_expr`, `add_instantiation`, `find`, etc.) canonicalize the `Id`s they return. Using `union_trusted` with the `Id`s returned by the internal variants of these methods seems to produce expected explanations.
eg.
```rust
    #[test]
    fn simple_explain_union_trusted() {
        crate::init_logger();
        let mut egraph = EGraph::new(()).with_explanations_enabled();

        let a = egraph.add_internal(SymbolLang::leaf("a"));
        let b = egraph.add_internal(SymbolLang::leaf("b"));
        let c = egraph.add_internal(SymbolLang::leaf("c"));
        let d = egraph.add_internal(SymbolLang::leaf("d"));
        egraph.union_trusted(a, b, "a=b");
        egraph.rebuild();
        let fa = egraph.add_internal(SymbolLang::new("f", vec![a]));
        let fb = egraph.add_internal(SymbolLang::new("f", vec![b]));
        egraph.union_trusted(c, fa, "c=fa");
        egraph.union_trusted(d, fb, "d=fb");
        egraph.rebuild();
        let exp = egraph.explain_equivalence(&"c".parse().unwrap(), &"d".parse().unwrap());
        println!("{:#}", exp.get_sexp());
    }
```
produces the explanation
```
(Explanation c (Rewrite=> c=fa (f a)) (f (Explanation a (Rewrite=> a=b b))) (Rewrite<= d=fb d))
```
but using `egraph.add` instead of `egraph.add_internal` produces
```
(Explanation c (Rewrite=> c=fa (f a)) (Rewrite<= d=fb d))
```

Is my understanding correct? Is the behaviour expected to be stable? If so, would it be worth making the internal variants of these functions public (and changing their names), and updating the documentation of `union_trusted`? 

After doing this it might also be nice to have a version of `explain_equivalence` that takes in (possibly non-canonical) `Id`s instead of `RecExpr`s and avoids calling `add_expr_internal`.

**Comment by oflatt** [comment]
This is a good summary of the current behavior when proofs are enabled. I'm in favor of making non-canonical `Id`s stable. I would be in favor of making your proposed change, so if you put up a PR I'll review it.

**Comment by dewert99** [comment]
Resolved by #283
